### PR TITLE
[Python] refactor slices on sorted

### PR DESCRIPTION
**Pull request by ScholliYT**
Sometimes you want to query the small element of a set of elements and use `sorted(elements)[0]` without a second thought. However, this is not optimal, since the entire list must be sorted first `O(n log n)`. It would be better to use the `min(elements)` method provided for this purpose `O(n)`. 
Furthermore `sorted(elements)[::-1]` is not very efficient, because it would be better to use `sorted(elements, reverse=True)` to save the slice operation.

**TLDR: using `sorted(elements)[0]` is slow and can be replaced with `min(elements)`.**

I stumbled across these code snippets while playing around with CodeQL (see https://lgtm.com/query/4148064474379348546/).

### CODE REVIEW

**Review by jansel — APPROVED** [review]

**Comment by ScholliYT** [comment]
@jansel Guess rebasing is broken because I edited one file that got changed in the meantime. I just merged the current origin/master into this branch. Is that fine or should I rebase my original commits onto origin/master?
